### ISSUE TITLE
ViT overfit ?

### ISSUE THREAD

**Issue by MJChku** [kentaroy47/vision-transformers-cifar10#4]
Do you know any theory for why vit overfit easily? 
Do you see situations when you vary parameters that vit overfits a lot ?
Thanks!

**Comment by kentaroy47** [comment]
I think vit is *underfitting* than overfitting.
Looks like if we pretrain ViT with a lot of images, you can get 99% with CIFAR 10.

**Comment by kentaroy47** [comment]
https://github.com/jeonsworld/ViT-pytorch
This repo gets 99% by using google's pretrained weights. 👀

**Comment by MJChku** [comment]
Yes if your vit dimension is large enough. using a small dimension, vit overfit to training data very quickly, which means it's powerful  enough to fit the data but overfit.

**Comment by zhoufengfan** [comment]
> I think vit is _underfitting_ than overfitting.
> Looks like if we pretrain ViT with a lot of images, you can get 99% with CIFAR 10.

I don't think the reason is underfitting. Because resnet can get a acc of about 90% when converging but vit cannot.

**Comment by kentaroy47** [comment]
![image](https://user-images.githubusercontent.com/39611941/128809999-23753cdb-c220-461b-80a1-f53d128e9793.png)

I added training with imagenet-pretrained-models to get 98% accuracy. Looks like pretraining is extremely important for ViT.
